pkg/inventory/inventory-client.go

```diff
@@ -36,6 +39,7 @@ type InventoryClient interface {
 	DeleteInventoryObj(inv *resource.Info) error
 	// SetDryRunStrategy sets the dry run strategy on whether this we actually mutate.
 	SetDryRunStrategy(drs common.DryRunStrategy)
+	ApplyInventoryNamespace(invNamespace *resource.Info) error
```

Comment:
Is this really the right place for this function? It doesn't seem like applying a namespace has any dependency on the inventory object.

Reply:
It's not a perfect fit, but I think this is the right place. This is related to the inventory object (since it is it's namespace). I don't think any other location would be especially better suited.